### PR TITLE
fix: Archived Conversations shouldn't fail tap

### LINKED ISSUE #36: Message sync failing due to deleted parent conversation
It seems like #22 didn't get rid of this error.

Exception:

```python
singer_sdk.exceptions.FatalAPIError: 410 Client Error: Gone for path: /v1/conversations/1d2f3f5136ed42b09eca9bc113fdfddb/messages . error_content={'errors': [{'code': 21, 'description': 'The conversation is deleted'}]}
```

### PR DESCRIPTION
**Pull request by visch**
Fixes #36

### CODE REVIEW

**Comment by visch** [comment]
Struggling to replicate this, I setup a situation where the conversation was archived when `conversation_message` was hit  and you get

```
{
    "count": 1,
    "items": [
        {
            "id": "feb9f5d69b004b528f24b85ca2f71260",
            "conversationId": "908b0f7764874b32b5e1fad0727bf713",
            "platform": "sms",
            "channelId": "7ce03f1645b64f8ab25930921b5c70b9",
            "type": "text",
            "content": {
                "text": "message is temporarily unavailable"
            },
            "status": "unknown",
            "createdDatetime": "2023-04-25T17:24:12.685968637Z",
            "updatedDatetime": "2023-04-25T17:24:12.685968637Z",
            "origin": "api"
        }
    ],
    "limit": 10,
    "offset": 0,
    "totalCount": 1
}
```

Back, I'm guessing this is due to not using the same `channel` as the bug. I'm trying to setup WhatApp on a demo account I have setup, and we'll see if I can get the same behavior.

@edgarrmondragon  I'm working through the precommit issues, but it's going to take some time as there's some ignores I think I should add and I"m not familiar with ruff and the top level "precommit checks" that happen outside of  the `tap-messagebird` repo itself.

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon I'm working through the precommit issues, but it's going to take some time as there's some ignores I think I should add and I"m not familiar with ruff and the top level "precommit checks" that happen outside of the `tap-messagebird` repo itself.

@visch I can send you a PR with fixes against this branch if that helps.

**Comment by visch** [comment]
> > @edgarrmondragon I'm working through the precommit issues, but it's going to take some time as there's some ignores I think I should add and I"m not familiar with ruff and the top level "precommit checks" that happen outside of the `tap-messagebird` repo itself.
> 
> @visch I can send you a PR with fixes against this branch if that helps.

I should be good, I just need to figure out how to add ignores now

**Comment by visch** [comment]
@edgarrmondragon  This is good to go PR wise. I'm trying to replicate this in my own test instance right now with WhatsApp as that's my gut feel into what the difference between our test ( https://github.com/MeltanoLabs/tap-messagebird/blob/main/scripts/generate_20_conversations.py with a breakpoint before archiving) and the failure is in production for this job

File: tap_messagebird/streams.py [hunk 1]
```diff
@@ -95,6 +99,40 @@ def get_url_params(
...
+        Args:
+            context: Stream partition or context dictionary.
+
+        Yields:
+            One item per (possibly processed) record in the API.
+        """
+        try:
+            yield from super().get_records(context)
+        except ConversationArchivedWarning as e:
+            self.logger.warning(f"Conversation is archived, skipping. {e=}")
```
Comment (edgarrmondragon) [inline tap_messagebird/streams.py:116]:
```suggestion
            self.logger.warning("Conversation is archived, skipping", exc_info=True)
```

File: pyproject.toml [hunk 2]
```diff
@@ -41,7 +41,7 @@ vcs = "git"
 style = "semver"
 
 [tool.ruff]
-ignore = ["ANN101"]
+ignore = ["ANN101", "G004", "PLR2004", "N818"]
```
Comment (edgarrmondragon) [inline pyproject.toml:44]:
```suggestion
ignore = ["ANN101", "PLR2004", "N818"]
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

File: tap_messagebird/streams.py [hunk 3]
```diff
@@ -95,6 +99,40 @@ def get_url_params(
...
+        """
+        if response.status_code == 410:
+            msg = (
+                f"{response.status_code} Client Error: "
+                f"{response.reason} for url: {response.url}"
+            )
+            response_json: dict = response.json()
+            error: dict = response_json["error"]
+            if response.status_code == 410 and error[0]["code"] == 21:
+                raise ConversationArchivedWarning(msg + f" {error=}")
```
Comment (edgarrmondragon) [inline tap_messagebird/streams.py:133]:
```suggestion
                errmsg = f"{msg} {error=}"
                raise ConversationArchivedWarning(errmsg)
```

File: tap_messagebird/streams.py (hunk 1 above)
Comment (visch) [reply tap_messagebird/streams.py:116]: til!

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**